drivers/awinic/aw20216.c

```diff
@@ -55,128 +53,79 @@
+    s_spi_transfer_buffer[0] = (AWINIC_ID | page | AW_WRITE);
+    s_spi_transfer_buffer[1] = reg;
 
-    if (spi_transmit(g_spi_transfer_buffer, 3) != SPI_STATUS_SUCCESS) {
+    if (spi_transmit(s_spi_transfer_buffer, 2) != SPI_STATUS_SUCCESS) {
         spi_stop();
         return false;
     }
-    spi_stop();
-    return true;
-}
 
-bool AW20216_init_scaling(void) {
-    // Set constant current to the max, control brightness with PWM
-    aw_led led;
-    for (uint8_t i = 0; i < DRIVER_LED_TOTAL; i++) {
-        led = g_aw_leds[i];
-        if (led.driver == 0) {
-            AW20216_write_register(DRIVER_1_CS, AW_PAGE_SCALING, led.r, AW_SCALING_MAX);
-            AW20216_write_register(DRIVER_1_CS, AW_PAGE_SCALING, led.g, AW_SCALING_MAX);
-            AW20216_write_register(DRIVER_1_CS, AW_PAGE_SCALING, led.b, AW_SCALING_MAX);
-        }
-#ifdef DRIVER_2_CS
-        else if (led.driver == 1) {
-            AW20216_write_register(DRIVER_2_CS, AW_PAGE_SCALING, led.r, AW_SCALING_MAX);
-            AW20216_write_register(DRIVER_2_CS, AW_PAGE_SCALING, led.g, AW_SCALING_MAX);
-            AW20216_write_register(DRIVER_2_CS, AW_PAGE_SCALING, led.b, AW_SCALING_MAX);
-        }
-#endif
+    if (spi_transmit(data, len) != SPI_STATUS_SUCCESS) {
+        spi_stop();
+        return false;
     }
+
+    spi_stop();
     return true;
 }
 
-bool AW20216_soft_enable(void) {
-    AW20216_write_register(DRIVER_1_CS, AW_PAGE_FUNCTION, AW_REG_CONFIGURATION, AW_CONFIG_DEFAULT | AW_CHIPEN);
-#ifdef DRIVER_2_CS
-    AW20216_write_register(DRIVER_2_CS, AW_PAGE_FUNCTION, AW_REG_CONFIGURATION, AW_CONFIG_DEFAULT | AW_CHIPEN);
-#endif
-    return true;
+static inline bool AW20216_write_register(pin_t cs_pin, uint8_t page, uint8_t reg, uint8_t value) {
+    // Little wrapper so callers need not care about sending a buffer
+    return AW20216_write(cs_pin, page, reg, &value, 1);
 }
 
-void AW20216_update_pwm(int index, uint8_t red, uint8_t green, uint8_t blue) {
-    aw_led led = g_aw_leds[index];
-    if (led.driver == 0) {
-        AW20216_write_register(DRIVER_1_CS, AW_PAGE_PWM, led.r, red);
-        AW20216_write_register(DRIVER_1_CS, AW_PAGE_PWM, led.g, green);
-        AW20216_write_register(DRIVER_1_CS, AW_PAGE_PWM, led.b, blue);
-    }
-#ifdef DRIVER_2_CS
-    else if (led.driver == 1) {
-        AW20216_write_register(DRIVER_2_CS, AW_PAGE_PWM, led.r, red);
-        AW20216_write_register(DRIVER_2_CS, AW_PAGE_PWM, led.g, green);
-        AW20216_write_register(DRIVER_2_CS, AW_PAGE_PWM, led.b, blue);
+static void AW20216_init_scaling(pin_t cs_pin) {
+    // Set constant current to the max, control brightness with PWM
+    for (uint8_t i = 0; i < AW_PWM_REGISTER_COUNT; i++) {
+        AW20216_write_register(cs_pin, AW_PAGE_SCALING, i, AW_SCALING_MAX);
     }
-#endif
-    return;
 }
 
-void AW20216_init(void) {
+void AW20216_init(pin_t cs_pin, pin_t en_pin) {
     // All LEDs should start with all scaling and PWM registers as off
```

Comment:
IMO this comment is a little unclear, the drivers should boot up with scaling and PWM registers set to 0, and then this init function sets up the per LED current scaling among other things. 
Probably just change `LEDs` to `drivers` or `driver ICs`